app/src/main/java/io/gnosis/safe/ui/safe/balances/coins/CoinBalanceAdapter.kt

```diff
@@ -33,9 +35,13 @@ class CoinBalanceViewHolder(private val viewBinding: ItemCoinBalanceBinding) : R
     fun bind(balanceItem: Balance) {
         with(viewBinding) {
             logoImage.loadTokenLogo(balanceItem.token.logoUrl)
+            val formatter = DecimalFormat.getInstance()
             symbol.text = balanceItem.token.symbol
-            balance.text = balanceItem.balance.shiftedString(balanceItem.token.decimals, 5)
-            balanceUsd.text = viewBinding.root.context.getString(R.string.usd_balance, balanceItem.balanceUsd)
+            balance.text = formatter.format(balanceItem.balance.shifted(balanceItem.token.decimals))
```

Comment:
this could be extracted to utility method

Reply:
Not exactly sure how do you mean that. `shifted` is already a utility method. Just passing its output to `format`

Reply:
the formatting part

Reply:
Can you provide a concrete suggestion? I don't see how I can meaningfully improve this by extracting a method. Then is more lines of code and passing the formatter around, if that is what you are suggesting.

Reply:
I suggest having formatting logic in one place. We could have formatting helper and inject it into adapter
```class BalanceFormatter(
val context: Context,
val formatter: DecimalFormatter
) {
	fun formatCurrency(balance: BigInteger, decimals: Int): String {
		return formatter.format(balance.shifted(decimals)
	}

	fun formatUsd(balanceUsd: BigDecimal): String {
	        return context.getString(R.string.usd_balance, formatter.fromat(balanceUsd.setScale(2, RoundingMode.DOWN))
	}
}```
This way currency formatting is encapsulated in one place and can be easily adjusted.

Reply:
can be done later though. Currently, balances are formatted only here

Reply:
That would definitely be the correct approach once we have to support various fiat currencies.